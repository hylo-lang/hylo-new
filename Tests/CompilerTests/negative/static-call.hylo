//! no-std

fun f0(x: Void) {}

fun f1<T>(x: T) {}

fun f2<T :: * -> *>() -> T { Builtin.trap() }

fun main() {
  f0<Void>(x: ())

  f1<Void>(x: ())
  f1<Void>(x: ((), ()))
  f1<Void, Void>(x: ())
}
