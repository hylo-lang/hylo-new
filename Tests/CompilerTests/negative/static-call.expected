negative/static-call.hylo:7.26-27: error: T expects 1 type argument(s)
fun f2<T :: * -> *>() -> T { Builtin.trap() }
                         ^
negative/static-call.hylo:10.3-5: error: 'f0' takes no compile-time arguments
  f0<Void>(x: ())
  ~~
negative/static-call.hylo:13.15-22: error: cannot pass value of type 'Void * Void' to parameter 'Void'
  f1<Void>(x: ((),()))
              ~~~~~~~
negative/static-call.hylo:14.3-5: error: 'f1' takes 1 compile-time argument(s); found 2
  f1<Void, Void>(x: ())
  ~~
